src/main/scala/pt/tecnico/dsi/neutron/services/Routes.scala

```diff
@@ -12,7 +12,7 @@ final class Routes[F[_] : Sync : Client](base: Uri, authToken: Header) extends S
 
   import dsl._
 
-  implicit val d: Decoder[List[Route]] = _.downField("router").get("routes")
+  implicit val d: Decoder[List[Route]] = _.downField("router").get("routes")(Decoder.decodeList(Route.codec))
```

Comment:
Curiosity: doesn't scala find the implicit decoder? Why do we need to explicitly give it?

Reply:
I don't know why but Scala allows a val to be implemented recursively, so in the right want side `d` is considered for the implicit search. That leads to stack overflow. By explicitly passing the decoder we break the recursive search.

Reply:
Makes sense. Imagine that you want to define a lazy never-ending list like fibonacci.